frontend/src/HomePage/Modal.jsx

```diff
@@ -31,6 +34,7 @@ export default function Modal({ title, show, closeModal, customClassName, childr
         ></button>
       </BootstrapModal.Header>
       <BootstrapModal.Body>{children}</BootstrapModal.Body>
+      {modalFooter ? modalFooter : <></>}
```

Comment:
You can just move that code here instead of using multiple conditions for same thing.